src/NzbDrone.Core/Notifications/Subsonic/SubsonicServerProxy.cs

```diff
@@ -0,0 +1,133 @@
+
+namespace NzbDrone.Core.Notifications.Subsonic
+{
+    public interface ISubsonicServerProxy
+    {
+        void Notify(SubsonicSettings settings, string message);
+        void Update(SubsonicSettings settings);
+        string Version(SubsonicSettings settings);
+    }
+
+    public class SubsonicServerProxy : ISubsonicServerProxy
+    {
+        private readonly Logger _logger;
+
+        public SubsonicServerProxy(Logger logger)
+        {
+            _logger = logger;
+        }
+
+        public void Notify(SubsonicSettings settings, string message)
+        {
+            var resource = "addChatMessage";
+            var request = GetSubsonicServerRequest(resource, Method.GET, settings);
+            request.AddParameter("message", message);
+            var client = GetSubsonicServerClient(settings);
+            var response = client.Execute(request);
+
+            _logger.Trace("Update response: {0}", response.Content);
+            CheckForError(response, settings);
+        }
+
+        public void Update(SubsonicSettings settings)
+        {
+            var resource = "startScan";
+            var request = GetSubsonicServerRequest(resource, Method.GET, settings);
+            var client = GetSubsonicServerClient(settings);
+            var response = client.Execute(request);
+
+            _logger.Trace("Update response: {0}", response.Content);
+            CheckForError(response, settings);
+        }
+
+        public string Version(SubsonicSettings settings)
+        {
+            var request = GetSubsonicServerRequest("ping", Method.GET, settings);
+            var client = GetSubsonicServerClient(settings);
+            var response = client.Execute(request);
+
+            _logger.Trace("Version response: {0}", response.Content);
+            CheckForError(response, settings);
+            
+            var xDoc = XDocument.Load(new StringReader(response.Content.Replace("&", "&amp;")));
+            var version = xDoc.Root?.Attribute("version")?.Value;
+
+            if (version == null)
+            {
+                throw new SubsonicException("Could not read version from Subsonic");
+            }
+
+            return version;
+        }
+
+        private RestClient GetSubsonicServerClient(SubsonicSettings settings)
+        {
+            var protocol = "http";
+
+            return RestClientFactory.BuildClient(string.Format("{0}://{1}:{2}/rest", protocol, settings.Host, settings.Port));
+        }
+
+        private RestRequest GetSubsonicServerRequest(string resource, Method method, SubsonicSettings settings)
+        {
+            var request = new RestRequest(resource, method);
+
+            if (settings.Username.IsNotNullOrWhiteSpace())
+            {
+                request.AddParameter("u", settings.Username);
+                request.AddParameter("p", settings.Password);
+                request.AddParameter("c", "Lidarr");
+                request.AddParameter("v", "1.15.0");
```

Comment:
What's this version for? Is it going to stay at this version?

Reply:
That defines what min version the server has to for our call to process. They just added the `startScan` api route in 1.15.0. http://www.subsonic.org/pages/api.jsp#startScan